### PR TITLE
ext_proc: Improved testing via a real gRPC server

### PR DESCRIPTION
**Pull request by gbrail**
Commit Message: Add an improved integration test for ext_proc
that uses an actual gRPC server

Additional Description: This lets us exercise ext_proc using actual
gRPC, which makes it possible to test buffering and streaming
cases that are harder to test using the current integration framework.
It works by starting an actual gRPC server in the background on
an ephemeral port which the filter talks to.

Risk Level: Low. This PR only adds new tests.

Signed-off-by: Gregory Brail <gregbrail@google.com>

### CODE REVIEW

File: test/extensions/filters/http/ext_proc/streaming_test.cc [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+using Http::LowerCaseString;
+
+// The buffer size for the listeners
+static const uint32_t BufferSize = 100000;
+
+// These tests exercise ext_proc using the integration test framework and a real gRPC server
+// for the external processor. This lets us more fully exercise all the things that happen
+// with larger, streamed payloads.
+
```
[the remark this answers is not in the document]
Comment (gbrail) [reply test/extensions/filters/http/ext_proc/streaming_test.cc]: Ack!

File: test/extensions/filters/http/ext_proc/streaming_test.cc [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+  Http::RequestEncoder&
+  sendClientRequestHeaders(absl::optional<std::function<void(Http::HeaderMap&)>> cb) {
+    auto conn = makeClientConnection(lookupPort("http"));
+    codec_client_ = makeHttpConnection(std::move(conn));
+    Http::TestRequestHeaderMapImpl headers;
+    HttpTestUtility::addDefaultHeaders(headers, std::string("POST"));
+    if (cb) {
+      (*cb)(headers);
+    }
+    auto enc_dec = codec_client_->startRequest(headers);
```
[the remark this answers is not in the document]
Comment (gbrail) [reply test/extensions/filters/http/ext_proc/streaming_test.cc]: Ack

File: test/extensions/filters/http/ext_proc/streaming_test.cc [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+using Http::LowerCaseString;
+
+// The buffer size for the listeners
+static const uint32_t BufferSize = 100000;
+
+// These tests exercise ext_proc using the integration test framework and a real gRPC server
+// for the external processor. This lets us more fully exercise all the things that happen
+// with larger, streamed payloads.
+
+class StreamingTest : public HttpIntegrationTest, public Grpc::GrpcClientIntegrationParamTest {
```
[the remark this answers is not in the document]
Comment (gbrail) [reply test/extensions/filters/http/ext_proc/streaming_test.cc]: Make sense -- thanks!

**Review by snowp — APPROVED** [review]
Thanks!

**Pull request merged**